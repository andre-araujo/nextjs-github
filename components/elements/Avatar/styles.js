import styled from 'styled-components';

export const Wrapper = styled.div`
    text-align: center;
`;

export const Image = styled.img`
    width: 100%;
    height: auto;
    border-radius: 5%;
`;
