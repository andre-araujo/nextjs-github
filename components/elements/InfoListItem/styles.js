import styled from 'styled-components';

export const Wrapper = styled.li`
    text-align: center;
    margin-bottom: 1rem;
`;

export const Title = styled.h3`
    font-family: AvantGardeMedium;
    font-size: 1.2rem;
    margin-right: .2rem;
    display: inline-block;
`;

export const Description = styled.p`
    font-family: AvantGardeLight;
    font-size: 1rem;
    display: inline-block;
`;
