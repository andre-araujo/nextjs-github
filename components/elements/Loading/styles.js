import styled, { css, keyframes } from 'styled-components';

const animateDot = keyframes`
    0% {
        opacity: 0;
        transform: rotate(-90deg);
    }

    50% {
        opacity: 1;
        transform: rotate(0);
    }
`;

export default styled.div`
    align-items: center;
    background: rgba(255, 255, 255, 0.85);
    bottom: 0;
    display: flex;
    left: 0;
    opacity: 0;
    pointer-events: none;
    position: fixed;
    justify-content: center;
    right: 0;
    top: 0;
    z-index: 10;

    ${({ active }) => active && css`
        opacity: 1;
        pointer-events: all;
    `}

    span {
        animation: ${animateDot} 1s infinite;
        background: #666;
        display: inline-flex;
        height: 1rem;
        margin: 0.5rem;
        opacity: 0;
        transform: rotate(90deg);
        width: 1rem;
    }

    span:nth-child(2) {
        animation-delay: 0.1s;
    }

    span:nth-child(3) {
        animation-delay: 0.2s;
    }
`;
