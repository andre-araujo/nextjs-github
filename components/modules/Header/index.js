import Link from 'next/link';
import { Box } from 'grid-styled';

import {
    Header,
    Wrapper,
    HeaderContent,
    Anchor,
} from './styles';

export default () => (
    <Wrapper>
        <Header>
            <HeaderContent>
                <Box
                    width={[1, '40rem', '50rem', '60rem']}
                    mx="auto"
                >
                    <Link href="/">
                        <Anchor>
                            â—€ Search
                        </Anchor>
                    </Link>
                </Box>
            </HeaderContent>
        </Header>
    </Wrapper>
);
