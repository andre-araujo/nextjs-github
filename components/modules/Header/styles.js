import styled from 'styled-components';

export const Wrapper = styled.div`
    height: 3rem;
    margin-bottom: 2rem;
`;

export const Header = styled.header`
    margin-bottom: 2rem;
    position: fixed;
    background: #fafafa;
    box-shadow: 0 0 .6rem rgba(0, 0, 0 ,.6);
    width: 100%;
    top: 0;
    left: 0;
`;

export const HeaderContent = styled.header`
    height: 3rem;
    display: flex;
    align-items: center;
    width: 90%;
    margin: 0 auto;
`;

export const Anchor = styled.a`
    text-decoration: none;
    color: #333;
    font-family: AvantGardeDemi;
    cursor: pointer;

    :hover {
        opacity: .6;
    }
`;
