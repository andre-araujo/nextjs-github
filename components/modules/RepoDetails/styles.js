import styled from 'styled-components';

export const Title = styled.h2`
    border-bottom: 1px solid #888;
    font-family: AvantGardeDemi;
    font-size: 2rem;
    line-height: 1;
    margin-bottom: .3em;
    padding-bottom: .3em;
    text-transform: uppercase;
`;

export const Label = styled.h3`
    font-family: AvantGardeMedium;
    font-size: 1.5rem;
    margin-bottom: .5rem;
    text-transform: capitalize;
`;

export const Description = styled.p`
    font-family: AvantGardeLight;
    font-size: 1.2rem;
    margin-right: .5rem;
`;
