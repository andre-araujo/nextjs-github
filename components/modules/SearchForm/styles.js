import styled from 'styled-components';

export const Form = styled.form`
    max-width: 30rem;
    margin: 0 auto;
    padding: 1rem;
    text-align: center;
`;

export const Row = styled.div`
    margin-bottom: 2rem;
`;
