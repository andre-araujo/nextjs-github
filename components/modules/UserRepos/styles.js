import styled from 'styled-components';

export default styled.h2`
    font-family: AvantGardeDemi;
    font-size: 2rem;
    line-height: 1;
    padding-bottom: .3em;
    margin-bottom: .3em;
    border-bottom: 1px solid #888;
    text-transform: uppercase;
`;
