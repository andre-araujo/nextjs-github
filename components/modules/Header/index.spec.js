import React from 'react';
import { mount } from 'enzyme';

import Header from './index';

import { Anchor } from './styles';

describe('Header component', () => {
    let routerMock;

    beforeEach(() => {
        routerMock = {
            back: jest.fn(),
        };
    });

    it('should match snapshot with Unavaliable description', () => {
        expect(mount(
            <Header
                router={routerMock}
            />,
        )).toMatchSnapshot();
    });

    it('should trigger back router method', () => {
        const header = mount(
            <Header
                router={routerMock}
            />,
        );

        header
            .find(Anchor)
            .simulate('click');

        expect(routerMock.back).toHaveBeenCalled();
    });
});
