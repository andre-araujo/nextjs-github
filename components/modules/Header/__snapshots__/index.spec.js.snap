// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot with Unavaliable description 1`] = `
<div
  className="sc-bxivhb doffTV"
>
  <header
    className="sc-ifAKCX hsZXKG"
  >
    <header
      className="sc-EHOje bNvVVQ"
    >
      <div
        className="sc-bdVaJa krDLuv"
        is={null}
      >
        <a
          className="sc-bZQynM CNreJ"
          onClick={[Function]}
        >
          â—€ Back
        </a>
      </div>
    </header>
  </header>
</div>
`;
